contracts/ColonyNetworkAuction.sol

```diff
@@ -0,0 +1,203 @@
+/*
+  This file is part of The Colony Network.
+
+  The Colony Network is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity ^0.4.17;
+pragma experimental "v0.5.0";
+pragma experimental "ABIEncoderV2";
+
+import "../lib/dappsys/math.sol";
+import "./ColonyNetworkStorage.sol";
+import "./ERC20Extended.sol";
+import "./IColony.sol";
+
+
+contract ColonyNetworkAuction is ColonyNetworkStorage {
+  event AuctionCreated(address auction, address token, uint256 quantity, uint256 startPrice);
+
+  function startTokenAuction(
+    address _token,
+    uint256 _quantity,
+    uint256 _startPrice)
+    public 
+    {
+    ERC20Extended clny = ERC20Extended(IColony(_colonies["Common Colony"]).getToken());
+    DutchAuction auction = new DutchAuction(address(clny), _token, _quantity, _startPrice);
+    ERC20Extended(_token).transfer(auction, _quantity);
+    AuctionCreated(address(auction), _token, _quantity, _startPrice);
+  }
+}
+
+
+// Dutch auction contract - distribution of a fixed number of tokens using an auction
+// The contract code is inspired by the Raiden and Gnosis Dutch auction contracts
+contract DutchAuction is DSMath {
+  address public colonyNetwork;
+  ERC20Extended public clnyToken;
+  ERC20Extended public token;
+
+  // Starting price in CLNY wei per Token wei
+  uint public startPrice;
+  uint public startBlock;
+  uint public endBlock;
+  // Keep track of all CLNY wei received
+  uint public receivedTotal;
+  // Keep track of cumulative CLNY funds for which the tokens have been claimed
+  uint public tokensClaimed;
+  // Total number of auctioned tokens
+  uint public quantity;
+  // CLNY Wei per Token wei
+  uint public finalPrice;
+  uint public constant MULTIPLIER = 10 ** 18;
```

Comment:
The comment above says start price startPrice is in CLNY wei and yet we're still multiplying by this factor in e.g. `price()`? The tests make it look like startPrice is CLNY. I *strongly* disagree with this `MULTIPLIER` being present at all - everything on-chain should be done in the indivisible unit, and clients deal with showing it in a 'human' form if desired.

Reply:
`MULTIPLIER` was meant to aid division precision not cater for wei. But since the start price will very high anyway, i.e. more than that multiplier catered for, it should be sufficient.